src/khoj/routers/api_chat.py

```diff
@@ -132,6 +137,60 @@ def chat_history(
     return {"status": "ok", "response": meta_log}
 
 
+@api_chat.get("/share/history")
+def chat_public_history(
+    request: Request,
+    common: CommonQueryParams,
+    public_conversation_slug: str,
+    n: Optional[int] = None,
+):
+    user = request.user.object if request.user.is_authenticated else None
+
+    # Load Conversation History
+    conversation = PublicConversationAdapters.get_public_conversation_by_slug(public_conversation_slug)
+
+    if conversation is None:
+        return Response(
+            content=json.dumps({"status": "error", "message": f"Conversation: {public_conversation_slug} not found"}),
+            status_code=404,
+        )
+
+    agent_metadata = None
+    if conversation.agent:
+        agent_metadata = {
+            "slug": conversation.agent.slug,
+            "name": conversation.agent.name,
+            "avatar": conversation.agent.avatar,
+            "isCreator": conversation.agent.creator == user,
+        }
+
+    meta_log = conversation.conversation_log
+    meta_log.update(
+        {
+            "conversation_id": conversation.id,
+            "slug": conversation.title if conversation.title else conversation.slug,
+            "agent": agent_metadata,
+        }
+    )
+
+    if n:
+        # Get latest N messages if N > 0
+        if n > 0 and meta_log.get("chat"):
+            meta_log["chat"] = meta_log["chat"][-n:]
+        # Else return all messages except latest N
+        elif n < 0 and meta_log.get("chat"):
+            meta_log["chat"] = meta_log["chat"][:n]
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="chat_history",
```

Comment:
api field doesn't match current api name
```suggestion
        api="get_chat_share",
```